OpenMTR/FileHandler.cs

```diff
@@ -0,0 +1,267 @@
+            {
+                if (!Directory.Exists(value)) throw new DirectoryNotFoundException("That directory does not exist, ensure it typed correctly and includeds the volume");
+                imagePath = standardizePath(value);
+                metadataPath = imagePath;
+                pos = 0;
+            }
+        }
+
+
+
+        /// <summary>
+        /// Change the directory to search for metadata files, if this is different from the image file directory, this will have to explicitly set
+        /// </summary>
+        public string MetadataDir
+        {
+            get
+            {
+                return metadataPath;
+            }
+            set
+            {
+                if (!Directory.Exists(value)) throw new DirectoryNotFoundException("That directory does not exist, ensure it typed correctly and includeds the volume");
+                metadataPath = standardizePath(value);
+            }
+        }
+
+
+
+        /// <summary>
+        /// Return the current data struct, read only
+        /// </summary>
+        public Files Data
+        {
+            get
+            {
+                return currentData;
+            }
+        }
+
+
+
+        /// <summary>
+        /// Return the full path of the current image
+        /// </summary>
+        public string ImageFile
+        {
+            get
+            {
+                return Dir + filenames[pos];
+            }
+        }
+
+
+
+        /// <summary>
+        /// Return the full path of the current metadata file
+        /// </summary>
+        public string MetadataFile
+        {
+            get
+            { // technically one line, broken out for readability
+                return
+                    MetadataDir +
+                    Regex.Replace(
+                        filenames[pos], extensions, "txt"
+                        );
+            }
+        }
+
+
+
+        //================================ STATIC METHODS ===========================================
+
+
+
+        /// <summary>
+        /// Converts a file to a bitmap, direct copy of Matt's method in the canny filter program
+        /// </summary>
+        /// <param name="fileName">The filename</param>
+        /// <returns>A bitmap</returns>
+        public static Bitmap ConvertToBitmap(string fileName)
+        {
+            Bitmap bitmap;
+            using (Stream bmpStream = File.Open(fileName, FileMode.Open))
+            {
+                Image image = Image.FromStream(bmpStream);
+
+                bitmap = new Bitmap(image);
+
+            }
+            return bitmap;
+        }
+
+
+
+        /// <summary>
+        /// Replace forward slashes with backslashes and add a trailing backslash if not present
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns>A standardized path</returns>
+        public static string standardizePath(string input)
+        {
+            string output = input.Replace('/','\\').Trim();

```

Comment:
Well........
Yeah, Windows paths are *supposed* to use \ and not /, but as it turns out all of these function still work if you do it the wrong way around. Using the Directory.Files() function above even returns a bunch of complete file paths with the wrong slash in them (it doesn't replace them with the correct characters). Admittedly if I did not have this, nothing really would break or change. But I wanted to be able to handle somebody inputting the wrong character because I didn't want to leave such strings unfixed. I dunno, I just wanted some input validation and none of the regular functions were really doing it for me. Should I change this?